ha/cli/exec/supportBundleExecutor.py

```diff
@@ -0,0 +1,152 @@
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import argparse
+import os
+
+from ha import const
+from ha.cli.exec.commandExecutor import CommandExecutor
+from cortx.utils.schema.payload import Tar
+from cortx.utils.process import SimpleProcess
+from ha.core.error import SupportBundleError
+from cortx.utils.log import Log
+
+class CreateExecutor(CommandExecutor):
+
+    def __init__(self):
+        """
+        This Class generates the Support Bundle for Component HA.
+        This method will generate bundle for HA and include the logs in it.
+        """
+        # TODO Add setup detection logic
+        self._id = None
+        self._path = None
+        self._args = None
+
+
+    def parse_cluster_args(self) -> bool:
+        '''
+           Parses the command line args.
+           Return: argparse
+        '''
+        parser = argparse.ArgumentParser(prog='support_bundle create')
+        parser.add_argument("support_bundle", help="Module name")
+        parser.add_argument("create", help="Operation")
+        parser.add_argument('--id', help='Bundle id', required=True)
+        parser.add_argument('--path', help='Bundle path', required=True)
+        self._args = parser.parse_args()
+        self._id = self._args.id
+        self._path = self._args.path
+        return True
+
+
+    def validate(self) -> str:
+        self.parse_cluster_args()
+        return True
+
+    def execute(self) -> str:
+        self.process_request()
+
+
+    def process_request(self, bundle_action="create"):
+        """
+        Process support Bundle request
+
+        Args:
+            action (string): Action to be taken for support bundle.
+            args (dict): Parameter for support bundle
+        """
+        getattr(self, bundle_action)()
+
+    def create(self):
+        """
+        Create support bundle on given path.
+        """
+        self._add_more_log()
+        export_path = self._export_logs(self._check_files())
+        print(f"Successfully created support bundle at {export_path}")
+        Log.info(f"Successfully created support bundle at {export_path}")
+
+    def _add_more_log(self):
+        """
+        Collect Extra Log
+        """
+        cmd_list = [const.PCS_STATUS, const.PCS_FAILCOUNT_STATUS]
+        with open(const.HA_CMDS_OUTPUT, "w") as file:
+            for cmd in cmd_list:
+                _err = ""
+                _proc = SimpleProcess(cmd)
```

Comment:
I think we use SimpleCommand()

Reply:
I have retained the pre-existing support bundle code as is. 
Maybe we can take up changes in the enhancement ticket EOS-17935 [CORTX-HA : Redesign support bundle generation] ??

Reply:
ok